### PR TITLE
[AMORO-2385] Make the maximum input file size for per optimize thread configurable

### LINKED ISSUE #2385: [Improvement]: Make the maximum input file size for per optimize thread configurable
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Currently, the maximum input file size for each optimize thread is fixed at 512MB. When processing tables with rapidly generated data, having a single optimize thread handle fewer files can cause the optimization merging speed to fall behind the rate at which the table is generating files. In cases where concurrent resources for optimization are sufficient, increasing the number of files processed by a single optimize thread can effectively address this issue.

### How should we improve?

Make the maximum input file size for per optimize thread configurable

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?

Close #2385 .

## Brief change log

- add config self-optimizing.max-input-file-size-per-thread

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (docs)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java [hunk 1]
```diff
@@ -128,7 +128,12 @@ public List<TaskDescriptor> planTasks() {
     // prioritize partitions with high cost to avoid starvation
     evaluators.sort(Comparator.comparing(PartitionEvaluator::getWeight, Comparator.reverseOrder()));
 
-    double maxInputSize = MAX_INPUT_FILE_SIZE_PER_THREAD * availableCore;
+    double maxInputSize =
```
[the remark this answers is not in the document]
Comment (zhongqishang) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java:131]: Thanks for the explanation, I understand. 👍

**Comment by zhoujinsong** [comment]
Supporting setting the maximum file size for each optimizer thread does make sense. This configuration further impacts the size of the process on the Optimizer. In practical scenarios, we have observed that when this value is too small, it results in minimal data being processed during each optimization, considering lengthy planning and ultimately causing the optimization speed to fall behind the table's write speed.

However, perhaps we should discuss where it would be more appropriate to set this configuration. Should it be a configuration of the optimizer or the table? If it's a configuration of the optimizer, maybe it would be more suitable for us to support it being set on the ams and optimizer group.

HDYT? @wangtaohz @hameizi

**Comment by wangtaohz** [comment]
Configuring at the _optimizer group_ or _AMS_ makes sense to me.

I suggest having a default configuration at the AMS level and an optional configuration at the optimizer group level, and the configuration in the optimizer group will override the AMS configuration.

Additionally, I believe that smaller granular configurations are unnecessary, like at the _optimizer_ level, because the tasks are not guaranteed to be executed by a specific optimizer.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**